### PR TITLE
Test/migrate controller tests to h2

### LINKED ISSUE #1321: Migrate webserver test from memory to H2
### Issue description

Since many endpoints are not implemented in the memory repositories, they cannot be tested.
We need to rewrite the test to work with a H2 database and add the missing tests

### PR DESCRIPTION
**Pull request by Skraye**
Will require this PR https://github.com/kestra-io/kestra/pull/1740 to be merged before tests pass.

close #1321

### CODE REVIEW

File: webserver/build.gradle [hunk 1]
```diff
@@ -26,5 +26,10 @@ dependencies {
     testImplementation project(':runner-memory')
```
Comment (loicmathieu) [inline webserver/build.gradle:26]:
```suggestion
    testImplementation project(':runner-memory')
```

This should be removed to prove it's working ;)

File: webserver/src/test/java/io/kestra/webserver/controllers/BlueprintControllerTest.java [hunk 2]
```diff
@@ -77,6 +76,7 @@ void blueprintFlow(WireMockRuntimeInfo wmRuntimeInfo) {
         wireMock.verifyThat(getRequestedFor(urlEqualTo("/v1/blueprints/id_1/flow")));
     }
 
+    @SuppressWarnings("unchecked")
```
Comment (loicmathieu) [inline webserver/src/test/java/io/kestra/webserver/controllers/BlueprintControllerTest.java:79]: Well, I prefere to keep those warnings but ... OK-ish

**Review by loicmathieu — CHANGES_REQUESTED** [review]
Did you check the impact on the test duration ?
Can you provide some numbers ?

File: webserver/src/test/java/io/kestra/webserver/controllers/BlueprintControllerTest.java (hunk 2 above)
Comment (Skraye) [reply webserver/src/test/java/io/kestra/webserver/controllers/BlueprintControllerTest.java:79]: We've used them over 100 times in the application, so I don't see the need for spam to tell us that we're doing unchecked cast unless we plan to correct it in the short term.

**Comment by Skraye** [comment]
> Did you check the impact on the test duration ? Can you provide some numbers ?

Well, it seems 10s faster on my machine, but I didn't try 100 times...

**Review by loicmathieu — CHANGES_REQUESTED** [review]

**Review by loicmathieu — APPROVED** [review]
LGTM

**Pull request merged**